### PR TITLE
Add support for other CI providers

### PR DESCRIPTION
**Pull request by sabrinabrogren**
PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep/semgrep/meta.py [hunk 1]
```diff
@@ -475,6 +475,254 @@ def to_dict(self) -> Dict[str, Any]:
         }
 
 
+@dataclass
+class CircleCIMeta(GitMeta):
+    """Gather metadata from Circle CI."""
+
+    environment: str = field(default="circle-ci", init=False)
```
Comment (underyx) [inline semgrep/semgrep/meta.py]:
```suggestion
    environment: str = field(default="circleci", init=False)
```

cuz they don't use a dash: https://github.com/circleci

File: semgrep/tests/e2e/test_ci.py [hunk 2]
```diff
@@ -222,6 +276,12 @@ def autofix(request, mocker):
         "github-pr",
         "gitlab",
         "gitlab-push",
+        "circle-ci",
```
Comment (underyx) [inline semgrep/tests/e2e/test_ci.py]:
```suggestion
        "circleci",
```

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines/meta.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+    "meta": {
+        "semgrep_version": "<sanitized version>",
+        "repository": "/project_name/project_name",
```
Comment (underyx) [inline semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines/meta.json]: the leading slash is probably not supposed to be here 🤔

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-bitbucket/meta.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+    "meta": {
+        "semgrep_version": "<sanitized version>",
+        "repository": "project_name/project_name",
+        "repo_url": "https://github.com/project_name/project_name.git",
+        "branch": "some/branch-name",
+        "ci_job_url": "https://github.com/project_name/project_name.git/addon/pipelines/home#!/results/a-uuid",
```
Comment (underyx) [inline semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-bitbucket/meta.json]: This CI job URL doesn't look like it'd work, considering it's under github.com

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-buildkite/meta.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+    "meta": {
+        "semgrep_version": "<sanitized version>",
+        "repository": "/project_name/project_name",
```
Comment (underyx) [inline semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-buildkite/meta.json]: scary leading slash again!

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines/meta.json (hunk 3 above)
Comment (sabrinabrogren) [reply semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines/meta.json]: Yeee this is the off by one error that I fixed (this comes from parsing the repo name from the github url)

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/noautofix-jenkins/meta.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+    "meta": {
+        "semgrep_version": "<sanitized version>",
+        "repository": "/org/repo",
```
Comment (underyx) [inline semgrep/tests/e2e/snapshots/test_ci/test_full_run/noautofix-jenkins/meta.json]: leading slash!

File: semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-bitbucket/meta.json (hunk 4 above)
Comment (sabrinabrogren) [reply semgrep/tests/e2e/snapshots/test_ci/test_full_run/autofix-bitbucket/meta.json]: oop nice catch, fixed!

**Review by underyx — APPROVED** [review]

**Pull request merged**